perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java

```diff
@@ -2179,75 +2179,153 @@ public Attribute call(ApiCaller ac, Deserializer parms) throws PerunException {
+	 *
+	 * @throw FacilityNotExistsException When the facility with <code>id</code> doesn't exist.
+	 * @throw WrongAttributeValueException When the attribute value is wrong/illegal.
+	 * @throw WrongAttributeAssignmentException When the attribute with <code>id</code> isn't facility attribute.
+	 * @throw WrongReferenceAttributeValueException When value of some Attribute is not correct regarding to other Attribute value.
+	 * @throw AttributeNotExistsException When the attribute with <code>id</code> doesn't exist.
 	 */
 	/*#
 	 * Checks if this vo attribute is valid.
 	 *
 	 * @param vo int Vo <code>id</code>
 	 * @param attribute int Attribute <code>id</code>
+	 *
+	 * @throw VoNotExistsException When the vo with <code>id</code> doesn't exist.
+	 * @throw WrongAttributeAssignmentException When the attribute isn't attribute of Vo with <code>id</code>.
+	 * @throw WrongAttributeValueException When the attribute value is wrong/illegal.
+	 * @throw WrongReferenceAttributeValueException When value of some Attribute is not correct regarding to other Attribute value.
+	 * @throw AttributeNotExistsException When the attribute with <code>id</code> doesn't exist.
 	 */
 	/*#
 	 * Checks if this member-resource attribute is valid.
 	 *
 	 * @param resource int Resource <code>id</code>
 	 * @param member int Member <code>id</code>
 	 * @param attribute int Attribute <code>id</code>
+	 *
+	 * @throw ResourceNotExistsException When the resource with <code>id</code> doesn't exist.
+	 * @throw MemberNotExistsException When the member with <code>id</code> doesn't exist.
+	 * @throw WrongAttributeValueException When the attribute value is wrong/illegal.
+	 * @throw WrongAttributeAssignmentException When the attribute with <code>id</code> isn't member-resource attribute.
+	 * @throw WrongReferenceAttributeValueException When value of some Attribute is not correct regarding to other Attribute value.
+	 * @throw AttributeNotExistsException When the attribute with <code>id</code> doesn't exist.
+	 * @throw MemberResourceMismatchException When the member with <code>id</code> and resource with <code>id</code> doesn't have the same Vo.
 	 */
 	/*#
 	 * Checks if this group-resource attribute is valid.
 	 *
 	 * @param resource int Resource <code>id</code>
 	 * @param group int Group <code>id</code>
 	 * @param attribute int Attribute <code>id</code>
+	 *
+	 * @throw AttributeNotExistsException When the attribute with <code>id</code> doesn't exist.
+	 * @throw ResourceNotExistsException When the resource with <code>id</code> doesn't exist.
+	 * @throw GroupNotExistsException When the group with <code>id</code> doesn't exist.
+	 * @throw WrongAttributeValueException When the attribute value is wrong/illegal.
+	 * @throw WrongAttributeAssignmentException When the attribute with <code>id</code> isn't group-resource attribute.
+	 * @throw GroupResourceMismatchException When the group with <code>id</code> and Resource with <code>id</code> doesn't have the same Vo.
```

Comment:
..maybe end it with ``aren't from the same Vo.`` Same for ``MemberResourceMismatchException``.